float32 time
float32 true_n
float32 true_velocity
float32 predicted_n
float32 predicted_velocity
float32 key_event
float32 key_event_time